libsolidity/formal/SMTEncoder.cpp

```diff
@@ -391,11 +391,20 @@ void SMTEncoder::endVisit(TupleExpression const& _tuple)
 	createExpr(_tuple);
 
 	if (_tuple.isInlineArray())
-		m_errorReporter.warning(
-			2177_error,
-			_tuple.location(),
-			"Assertion checker does not yet implement inline arrays."
-		);
+	{
+		createExpr(_tuple);
+
+		// Add constraints for the length and values as it is known.
+		auto symbArray = dynamic_pointer_cast<smt::SymbolicArrayVariable>(m_context.expression(_tuple));
+		solAssert(symbArray, "");
+
+		auto const& values = _tuple.components();
+		m_context.addAssertion(symbArray->length() == values.size());
+		for (size_t i = 0; i < values.size(); i++)
+			m_context.addAssertion(
+				smtutil::Expression::select(symbArray->elements(), i) == expr(*values[i])
+			);
```

Comment:
This piece of code is almost the same like in String literal case in `endVisit(Literal)`
Will create shared helper function that does work for both.